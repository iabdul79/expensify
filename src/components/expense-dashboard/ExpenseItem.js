import React from 'react'

export default (props) => {
  const {description, amount, createdAt} = props.expense;
  return (
    <div>
      <h2>{description}</h2>
      <p>{amount} - {createdAt}</p>
  </div>
  )
}