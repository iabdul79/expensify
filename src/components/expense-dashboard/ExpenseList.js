import React from 'react'
import { connect } from 'react-redux'
import { getVisibleExpense } from '../../selectors/expense-selectors'
import ExpenseItem from './ExpenseItem'

export const ExpenseList = (props) => (
  <div className="main-container">
    <div className="list__header">
      <div className="for-mobile">Expense</div>
      <div className="for-desktop">Expense</div>
      <div className="for-desktop">Amount</div>
    </div>
    <div className="list__body">
      {
        props.expenses.length ?
        props.expenses.map(expense => <ExpenseItem key={expense.id} expense={expense} />) : 
        <div className="list-item list-item__message">
          <span> No Expenses Available</span>
        </div>
      }
    </div>
  </div>
)

const stateMapper = (state) => ({
  expenses: getVisibleExpense(state.expenses, state.filters)
})
const connector = connect(stateMapper)

export default connector(ExpenseList)