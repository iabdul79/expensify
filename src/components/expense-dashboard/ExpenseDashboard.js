import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilter from '../ExpenseListFilter'
import ExpenseSummary from '../ExpenseSummary'

export default () => (
  <div>
    <ExpenseListFilter />
    <ExpenseSummary />
    <ExpenseList />
  </div>
)