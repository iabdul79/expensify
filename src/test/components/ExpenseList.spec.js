import React from 'react'
import {shallow} from 'enzyme'
import {ExpenseList} from '../../components/expense-dashboard/ExpenseList'
import expenses from '../fixtures/expenses'

test('should render Expense list component', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />)
  expect(wrapper).toMatchSnapshot()
})

test('should show no expense available on expense list component test', () => {
  const wrapper = shallow(<ExpenseList expenses={[]}/>)
  expect(wrapper).toMatchSnapshot()
})