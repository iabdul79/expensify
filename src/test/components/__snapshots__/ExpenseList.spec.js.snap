// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense list component 1`] = `
<div
  className="main-container"
>
  <div
    className="list__header"
  >
    <div
      className="for-mobile"
    >
      Expense
    </div>
    <div
      className="for-desktop"
    >
      Expense
    </div>
    <div
      className="for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list__body"
  >
    <_default
      expense={
        Object {
          "amount": 120,
          "createdAt": 0,
          "description": "Gum",
          "id": "1",
          "note": "",
        }
      }
      key="1"
    />
    <_default
      expense={
        Object {
          "amount": 120300,
          "createdAt": -345600000,
          "description": "Rent",
          "id": "2",
          "note": "",
        }
      }
      key="2"
    />
    <_default
      expense={
        Object {
          "amount": 12100,
          "createdAt": 345600000,
          "description": "Credit card",
          "id": "3",
          "note": "",
        }
      }
      key="3"
    />
  </div>
</div>
`;

exports[`should show no expense available on expense list component test 1`] = `
<div
  className="main-container"
>
  <div
    className="list__header"
  >
    <div
      className="for-mobile"
    >
      Expense
    </div>
    <div
      className="for-desktop"
    >
      Expense
    </div>
    <div
      className="for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list__body"
  >
    <div
      className="list-item list-item__message"
    >
      <span>
         No Expenses Available
      </span>
    </div>
  </div>
</div>
`;
